presto-tests/src/test/resources/resource_groups_query_type_based_config.json

```diff
@@ -44,6 +44,12 @@
           "hardConcurrencyLimit": 3,
           "maxQueued": 4
         },
+        {
+          "name": "transaction",
```

Comment:
[major] This reminds me that the rollout of this commit will cause pretty big production regression. We are now going to query queries like `set session`. Do we have a safe rollout strategy?

Reply:
Good catch! We will make changes to the existing resource groups to account for this query type. We will make sure to account this for the next release.